Fix Autocomplete selection display with custom Component (#4367)
marmelab/react-admin: docs/Inputs.md

<AutocompleteInput source="author_id" choices={choices} optionText={optionRenderer} />
```

> `optionText` also accepts a custom Component. If you do so, you will have to set the inputText function wich determine the input text filter of the current selection:

Reviewer: ```suggestion
`optionText` also accepts a custom Component. If you provide one, you will have to set the `inputText` function wich determine the input text filter of the current selection:
```

> `optionText` also accepts a custom Component. If you provide one, you will have to set the `inputText` function wich determine the input text filter of the current selection:

Reviewer 2: ```suggestion
`optionText` also accepts a custom Component. However, as the underlying Autocomplete component requires that the current selection is a string, if you opt for a Component, you must pass a function as the `inputText` prop. This function should return text representation of the current selection:
```

| `emptyValue` | Optional | anything | `''` | The value to use for the empty element |
| `emptyText` | Optional | `string` | `''` | The text to use for the empty element |
| `matchSuggestion` | Optional | `Function` | - | Required if `optionText` is a React element. Function returning a boolean indicating whether a choice matches the filter. `(filter, choice) => boolean`
| `optionText` | Optional | <code>string &#124; Function &#124; Component</code> | `name` | Fieldname of record to display in the suggestion item or function which accepts the correct record as argument (`(record)=> {string}`) |
| `optionValue` | Optional | `string` | `id` | Fieldname of record containing the value to use as input value  |

> | `inputText` | Optional | <code>Function</code> | `-` | If `optionText` is a custom Component, This function is needed to determine the text displayed filter of the current selection. In Autocomplete, the current selection have to be an editable string and can not be a Component  (`(record)=> {string}`) |

Reviewer 2: ```suggestion
| `inputText` | Optional | <code>Function</code> | `-` | If `optionText` is a custom Component, this function is needed to determine the text displayed for the current selection. |
```
